go-controller/pkg/clustermanager/routeadvertisements/controller.go

```diff
@@ -280,6 +280,183 @@ func (c *Controller) reconcile(name string) error {
 	return c.updateRAStatus(ra, hadUpdates, err)
 }
 
+func subnetsOverlap(subnet1, subnet2 string) (bool, error) {
+	ipNets, err := util.ParseIPNets([]string{subnet1, subnet2})
+	if err != nil {
+		return false, err
+	}
+
+	// Two subnets overlap if one contains the other's IP or vice versa
+	return ipNets[0].Contains(ipNets[1].IP) || ipNets[1].Contains(ipNets[0].IP), nil
+}
+
+// checkSubnetOverlaps validates that subnets from the current RA don't overlap
+// with any existing subnets in the same VRFs, both within the RA itself and
+// across different RAs
+func (c *Controller) checkSubnetOverlaps(ra *ratypes.RouteAdvertisements, selectedNetworks *selectedNetworks) error {
+	vrfSubnets := createVRFSubnetsMap(ra, selectedNetworks)
+
+	// Check for overlaps within each VRF
+	for vrf, subnets := range vrfSubnets {
+		// Check for overlaps within current RA
+		if err := checkWithinRAOverlaps(subnets); err != nil {
+			return fmt.Errorf("%w: error checking subnet overlap on RouteAdvertisement %q in VRF %q: %w", errConfig, ra.Name, vrf, err)
+		}
+		// Check for overlaps between current RA and other RAs in the same VRF
+		if err := c.checkCrossRAOverlaps(ra, vrf, subnets); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+// checkWithinRAOverlaps checks if subnets within the current RA overlap with each other
+func checkWithinRAOverlaps(subnets []string) error {
```

Comment:
This has nothign to do with RA is just subnet slice check

```suggestion
func checkSubnetSliceOverlaps(subnets []string) error {
```

Reply:
DONE